mne/io/boxy/boxy.py

```diff
@@ -527,6 +533,90 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+                        
+                        print('Removing phase mean')
+                        # scipy.io.savemat(file_name = r"C:\Users\spork\Desktop\data_detrend_matlab.mat",
+                        #               mdict=dict(data=data_))
+                        
+                        mrph = np.mean(data_,axis=1);
+                        for i_chan in range(np.size(data_, axis=0)):
+                            data_[i_chan,:]=(data_[i_chan,:]-mrph[i_chan])
+                            
+                        # mean_data = scipy.io.loadmat(r"C:\Users\spork\Desktop\data_mean_python.mat")
+                        
+                        # test3 = abs(mean_data['data'] - data_) <= thresh
+                        # test3.all()
+                        
+                        print('Removing phase outliers')
+                        # scipy.io.savemat(file_name = r"C:\Users\spork\Desktop\data_mean_matlab.mat",
+                        #               mdict=dict(data=data_))
+                        
+                        ph_out_thr=3; # always set to "3" per Kathy & Gabriele Oct 12 2012
+                        sdph=np.std(data_,1, ddof = 1); #set ddof to 1 to mimic matlab
+                        n_ph_out = np.zeros(np.size(data_, axis=0), dtype= np.int8)
+                        
+                        for i_chan in range(np.size(data_, axis=0)):
+                            outliers = np.where(np.abs(data_[i_chan,:]) > 
+                                        (ph_out_thr*sdph[i_chan]))
+                            outliers = outliers[0]
+                            if len(outliers) > 0:
+                                if outliers[0] == 0:
+                                    outliers = outliers[1:]
+                                if len(outliers) > 0:
+                                    if outliers[-1] == np.size(data_, axis=1) - 1:
+                                        outliers = outliers[:-1]
+                                    n_ph_out[i_chan] = int(len(outliers))
+                                    for i_pt in range(n_ph_out[i_chan]):
+                                        j_pt = outliers[i_pt]
+                                        data_[i_chan,j_pt] = (
+                                            (data_[i_chan,j_pt-1] + 
+                                              data_[i_chan,j_pt+1])/2)
+                                        
+                        # outlier_data = scipy.io.loadmat(r"C:\Users\spork\Desktop\data_outliers_python.mat")
+                        
+                        # test4 = abs(outlier_data['data'] - data_) <= thresh
+                        # test4.all()
+                        
+                        #convert phase to pico seconds
+                        for i_chan in range(np.size(data_, axis=0)):
+                            data_[i_chan,:] = ((1e12*data_[i_chan,:])/
+                                               (360*mtg_mdf[i_mtg][i_chan]))
```

Comment:
would this work for files with multiple modulation frequencies?

Reply:
this means we want different data types with their own scaling factors (for plotting only, doesn't change the data scale) in the defaults.py file, one for phase and one for AC (we can probably ignore DC signals)

Reply:
it should work with multiple modulation frequencies since it does keep track of the modulation frequency for each channel/row in the mtg file.

So add something like frnirs_ac and fnirs_ph to default.py? What scaling factors would we use?

Reply:
great

yeah we should add those two new data types exactly, probably need to play around with the scaling factors and see what is a good range